From d0bd65b0e2f3c1319b4973af27410fd4c13aa00e Mon Sep 17 00:00:00 2001
From: Akihiro Sagawa <sagawa.aki@gmail.com>
Date: Tue, 4 Nov 2025 21:14:12 +0900
Subject: [PATCH] winegstreamer: Always use bottom-up for AVI RGB videos.

GStreamer caps don't provide the scan direction of AVI files,
and some downstream filters don't handle top-down correctly.

diff --git a/dlls/winegstreamer/quartz_parser.c b/dlls/winegstreamer/quartz_parser.c
index d8462770..6ba0b212 100644
--- a/dlls/winegstreamer/quartz_parser.c
+++ b/dlls/winegstreamer/quartz_parser.c
@@ -2432,6 +2432,8 @@ static HRESULT avi_splitter_source_query_accept(struct parser_source *pin, const
     HRESULT hr;
 
     wg_parser_stream_get_current_format(pin->wg_stream, &format);
+    if (wg_video_format_is_rgb(format.u.video.format))
+        format.u.video.height = -format.u.video.height;
     if (!amt_from_wg_format(&pad_mt, &format, false))
         return E_OUTOFMEMORY;
     hr = compare_media_types(mt, &pad_mt) ? S_OK : S_FALSE;
@@ -2471,6 +2473,8 @@ static HRESULT avi_splitter_source_get_media_type(struct parser_source *pin,
     else if (index > 0)
         return VFW_S_NO_MORE_ITEMS;
 
+    if (wg_video_format_is_rgb(format.u.video.format))
+        format.u.video.height = -format.u.video.height;
     if (!amt_from_wg_format(mt, &format, false))
         return E_OUTOFMEMORY;
     return S_OK;
